### PR TITLE
torch-1.9.0 + various version upgrades

### PR DESCRIPTION
**Pull request by azhou-determined**
## Description
**Versions upgraded:**
torch: 1.7.1 -> 1.9.0
torchvision: 0.8.2 -> 0.10.0
	- to support PyTorch profiler and CUDA 11
horovod: 0.22.0 -> 0.22.1
        - fixes for sparse gradients in torch 1.8.1+
tensorflow: 2.4.1 -> 2.4.2
	- various security fixes
pytorch_lightning: 1.2.0 -> 1.3.5
	- upgrade pyyaml to 5.4.1 to fix security error preventing images from publishing on Nvidia registry

- Added README and .gitignore

**Testing**
image builds:
- andazhou/environments:py-3.7-pytorch-1.9-lightning-1.3-tf-2.4-cpu-0.16.0
- andazhou/environments:cuda-11.1-pytorch-1.9-lightning-1.3-tf-2.4-gpu-0.16.0

## Checklist

- [ ] Bump VERSION to make the pushed images are tagged with the right version.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: Dockerfile.gpu [hunk 1]
```diff
@@ -44,7 +44,11 @@ ARG TENSORFLOW_PIP
 ARG TORCH_PIP
 ARG TORCHVISION_PIP
 ARG LIGHTNING_PIP
+ARG TF_CUDA_SYM
+
 RUN if [ "$TENSORFLOW_PIP" ]; then python -m pip install $TENSORFLOW_PIP; fi
+RUN if [ "$TF_CUDA_SYM" ]; then /tmp/det_dockerfile_scripts/tf2.4-cuda11.1-symlink.sh; fi
```
Comment (azhou-determined) [inline Dockerfile.gpu:50]: explanation for this: cuda 11.0 uses a math package cusolver 10.0, cuda 11.1 comes with cusolver 11. tensorflow 2.4 is looking for cusolver 10.0. putting a soft link between the LD_LIBRARY_PATH cusolver and tensorflow's cusolver dependency fixes the issue. cusolver 11 seems to be backwards compatible with cusolver 10. there doesn't appear to be any issues so far.

File: Makefile [hunk 2]
```diff
@@ -9,14 +9,15 @@ CPU_PREFIX := environments:py-3.7-
 CPU_SUFFIX := -cpu
 CUDA_102_PREFIX := environments:cuda-10.2-
 CUDA_110_PREFIX := environments:cuda-11.0-
```
Comment (ioga) [inline Makefile:11]: this variable seems to be no longer in use - remove it?

File: README.md [hunk 3]
```diff
@@ -1 +1,26 @@
...
+
+###Run
+To build a custom image:
+- Modify an existing recipe in `Makefile` or define a new command.
+- Add/modify any additional scripts needed for the image build
+- run `make {build_name}`
+
+To publish an image:
+- Add a `publish` target or modify an existing recipe
+- Change registry info (`DOCKERHUB_REGISTRY` and `NGC_REGISTRY`) in `Makefile`
```
Comment (ioga) [inline README.md:25]: this can also be done on a command line, e.g. `make build-tf25-gpu DOCKERHUB_REGISTRY=determinedilia`

File: dockerfile_scripts/tf2.4-cuda11.1-symlink.sh [hunk 4]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+
+set -e
+
+ln -s /usr/local/cuda/lib64/libcusolver.so.11 /opt/conda/lib/python3.7/site-packages/tensorflow/python/libcusolver.so.10
```
Comment (ioga) [inline dockerfile_scripts/tf2.4-cuda11.1-symlink.sh:5]: this is a style preference, but I'd inline such one-liners directly into dockerfile. a separate file feels redundant and doesn't improve readability that much.

**Review by ioga — APPROVED** [review]
lgtm

**Comment by ioga** [comment]
can you also please bump the `VERSION` file number?

File: dockerfile_scripts/tf2.4-cuda11.1-symlink.sh (hunk 4 above)
Comment (azhou-determined) [reply dockerfile_scripts/tf2.4-cuda11.1-symlink.sh]: good point. created it thinking there would be more symlinking needed but this is unnecessary now. removed

**Review by ioga — APPROVED** [review]
lgtm, thank you for taking care of the upgrades

**Review by ioga — APPROVED** [review]
